lib/ruby_lsp/document.rb

```diff
@@ -32,6 +32,13 @@ def ==(other)
       @source == other.source
     end
 
+    sig { void }
+    def parse
+      @tree = SyntaxTree.parse(@source)
+    rescue SyntaxTree::Parser::ParseError
+      # Do not raise if we failed to parse
```

Comment:
Should we log something in the console in this case?

Reply:
I think it's valuable, but I'll suggest doing this in a follow up PR. If we're going to log something, we should include the URI, which is currently not passed to `Document`. I can create a follow up as soon as this one goes in.